confidant/scripts/archive.py

```diff
@@ -0,0 +1,169 @@
+        ]
+        for credential in credentials:
+            if self.credential_in_service(credential.id, services):
+                msg = ('Skipping archival of disabled credential {}, as it'
+                       ' is still mapped to a service.')
+                logger.warning(msg.format(credential.id))
+                continue
+            saves = []
+            deletes = []
+            # save the current record.
+            archive_credential = CredentialArchive.from_credential(
+                credential,
+            )
+            saves.append(archive_credential)
+            # fetch and save every revision
+            _range = range(1, credential.revision + 1)
+            ids = []
+            for i in _range:
+                ids.append("{0}-{1}".format(credential.id, i))
+            revisions = Credential.batch_get(ids)
+            for revision in revisions:
+                archive_revision = CredentialArchive.from_credential(
+                    revision,
+                )
+                saves.append(archive_revision)
+                deletes.append(revision)
+            deletes.append(credential)
+            try:
+                self.save(saves, force=force)
+            except Exception:
+                logger.exception(
+                    'Failed to batch save {}, skipping deletion.'.format(
+                        credential.id
+                    )
+                )
+                continue
+            try:
+                self.delete(deletes, force=force)
+            except Exception:
+                logger.exception(
+                    'Failed to batch delete {}'.format(credential.id)
+                )
+
+    def run(self, days, force, ids):
+        if not settings.DYNAMODB_TABLE_ARCHIVE:
+            logger.error('DYNAMODB_TABLE_ARCHIVE is not configured, exiting.')
+            return 1
+        if days and ids:
+            logger.error('--days and --ids options are mutually exclusive')
+            return 1
+        if not days and not ids:
+            logger.error('Either --days or --ids options are required')
+            return 1
+        credentials = []
+        if ids:
+            # filter strips an empty string
+            _ids = [_id.strip() for _id in list(filter(None, ids.split(',')))]
+            if not _ids:
+                logger.error('Passed in --ids argument is empty')
+                return 1
+            for credential in Credential.batch_get(_ids):
+                if credential.enabled:
+                    logger.warning(
+                        'Skipping enabled credential {}'.format(credential.id)
+                    )
+                    continue
+                credentials.append(credential)
+        else:
+            for credential in Credential.data_type_date_index.query(
+                'credential'
+            ):
+                tz = credential.modified_date.tzinfo
+                now = datetime.now(tz)
```

Comment:
Not necessary but a slight optimization would be to move these 2 lines outside the loop since tzinfo is always UTC time (thanks to pynamo)